src/internal/utils.js

```diff
@@ -19,6 +19,11 @@ export function check(value, predicate, error) {
   }
 }
 
+const hasOwnProperty = Object.prototype.hasOwnProperty
+export function hasProperty(object, property) {
```

Comment:
I went ahead and just added this to utils as an export so it can be reused if needed down the road. If you think this unnecessary I can change it.

It annoys me that I am using `is.notUndef` before the object `is` gets defined in the file. but if I put the helper _below_ the definition of `is`, it would be falling `hasProperty` above the definition of it. I hate moments like this.